sql/auth/sql_authentication.cc

```diff
@@ -1382,9 +1383,7 @@ static bool send_server_handshake_packet(MPVIO_EXT *mpvio, const char *data,
 
   end = my_stpnmov(end, server_version, SERVER_VERSION_LENGTH);
   end = my_stpcpy(end, " ");
-  end = my_stpnmov(end, MYSQL_COMPILATION_COMMENT,
-                   SERVER_VERSION_LENGTH - (end - buff - 1)) +
-        1;
+  end = my_stpcpy(end, MYSQL_COMPILATION_COMMENT) + 1;
```

Comment:
@hermanlee 
Probably out of scope for this fix but should we use `server_version_comment_ptr` instead of `MYSQL_COMPILATION_COMMENT` here?

Reply:
Doesn't matter here because `Sys_version_comment` is READ_ONLY with `DEFAULT(MYSQL_COMPILATION_COMMENT)`. It matters however for PS.